### ISSUE TITLE
No compliant error 

### ISSUE THREAD

**Issue by sihui-mr** [stylelint/stylelint#5595]
> style lint can catch but no throw node error 

> What code is needed to reproduce the bug?

```    "stylelint": "^13.13.1",
    "stylelint-config-property-sort-order-smacss": "^7.1.0",
    "stylelint-config-standard": "^22.0.0",
    "stylelint-config-styled-components": "^0.1.1",
```

```css
[class^='abcd'] {
      overflow-x: hidden;


      
    }
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```javascript
module.exports = {
  extends: [
    'stylelint-config-standard',
    'stylelint-config-styled-components',
    'stylelint-config-property-sort-order-smacss',
  ],
  plugins: [
    'stylelint-order',
  ],
};

```

> Which version of stylelint are you using?

`13.13.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

` "stylelint $(git diff --name-only --diff-filter=ACMRTUXB origin/release | grep -E \"(.ts$|.tsx$)\") || echo \".stylelintrc.js\" ",`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?
No, styled component 

> What did you expect to happen?

throw error 

> What actually happened (e.g. what warnings or errors did you get)?

![Screen Shot 2021-10-15 at 4 05 43 pm](https://user-images.githubusercontent.com/90160012/137435385-d4f014cf-5d1d-42be-bbf3-f904c59ca6c9.png)

**Comment by jeddy3** [comment]
Please create a minimum reproducible repo to help us better understand the issue you're having.
